Sources/FluentKit/Properties/Siblings.swift

```diff
@@ -117,20 +117,24 @@ public final class SiblingsProperty<From, To, Through>
         _ edit: (Through) -> () = { _ in }
     ) -> EventLoopFuture<Void> {
         guard let fromID = self.idValue else {
-            fatalError("Cannot attach siblings relation \(self.name) to unsaved model.")
+            return database.eventLoop.makeFailedFuture(SiblingsPropertyError.owningModelIdRequired(property: self.name))
         }
-
-        return tos.map { to -> Through in
+        
+        var pivots: [Through] = []
+        pivots.reserveCapacity(tos.count)
```

Comment:
I never went ahead to look into `map`'s implementation but i always assumed it must already do the `reserveCapacity` itself? Does it not do that?! If yes, do you know why?!

Reply:
It absolutely does. But that has nothing to do with why I didn't use `map()` here anyhow. Think about what the loop body has to deal with - this an old-style ELF implementation, remember; it can't just `throw` to terminate a `map()`.

Reply:
👌